### PR TITLE
Fix series status parsing

### PR DESCRIPTION
**Pull request by Shadowghost**
NFOs can include more than our three status descriptions: https://www.themoviedb.org/bible/tv

**Changes**
* Add helper function to properly transform different strings into our enum values

### CODE REVIEW

File: Emby.Naming/TV/TvParserHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    /// <returns>Returns true if parsing was successful.</returns>
+    public static bool TryParseSeriesStatus(string status, out SeriesStatus? enumValue)
+    {
+        if (Enum.TryParse(status, true, out SeriesStatus seriesStatus))
+        {
+            enumValue = seriesStatus;
+            return true;
+        }
+
+        switch (status)
```
Comment (crobibero) [inline Emby.Naming/TV/TvParserHelpers.cs:26]: switch is case-sensitive, should probably do case-insensitive matching here

File: Emby.Naming/TV/TvParserHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace Emby.Naming.TV;
+
+/// <summary>
+/// Helper class for TV metadata parsing.
+/// </summary>
+public static class TvParserHelpers
+{
+    private static readonly List<string> _continuingState = ["Pilot", "Returning Series", "Returning"];
+    private static readonly List<string> _endedState = ["Cancelled"];
```
Comment (Bond-009) [inline Emby.Naming/TV/TvParserHelpers.cs:14]: Use arrays instead of lists

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: tests/Jellyfin.Naming.Tests/TV/TvParserHelpersTest.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+using Emby.Naming.TV;
+using MediaBrowser.Model.Entities;
+using Xunit;
+
+namespace Jellyfin.Naming.Tests.TV
```
Comment (Bond-009) [inline tests/Jellyfin.Naming.Tests/TV/TvParserHelpersTest.cs:5]: Use file scope namespaces for new files

File: tests/Jellyfin.Naming.Tests/TV/TvParserHelpersTest.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        [InlineData("Cancelled", SeriesStatus.Ended, true)]
+        [InlineData("Continuing", SeriesStatus.Continuing, true)]
+        [InlineData("Returning", SeriesStatus.Continuing, true)]
+        [InlineData("Returning Series", SeriesStatus.Continuing, true)]
+        [InlineData("Unreleased", SeriesStatus.Unreleased, true)]
+        [InlineData("XXX", null, false)]
+        public void SeriesStatusParserTest(string statusString, SeriesStatus? status, bool success)
+        {
+            var successful = TvParserHelpers.TryParseSeriesStatus(statusString, out var parsered);
+            Assert.Equal(success, successful);
```
Comment (Bond-009) [inline tests/Jellyfin.Naming.Tests/TV/TvParserHelpersTest.cs:20]: I prefer separate tests for valid and invalid inputs, it reduces the complexity of the test function and reduces arguments

**Review by Bond-009 — APPROVED** [review]

**Review by crobibero — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**